database/sinkdb/state.go

```diff
@@ -90,6 +90,13 @@ func (s *state) RestoreSnapshot(data []byte, index uint64) error {
 	s.peers = snapshot.Peers
 	s.state = snapshot.State
 	s.version = snapshot.Version
+
+	if s.version == nil {
+		s.version = make(map[string]uint64)
+	}
+	if s.peers == nil {
+		s.peers = make(map[uint64]string)
+	}
```

Comment:
> 2. a snapshot should always at least have a version for the nextNodeID.

Ah sorry, I forgot Chain Core 1.2.x saves snapshots without any version data, so the nil check for s.version is important as long as we support migrating from Chain Core 1.2.x. Maybe keep that check but add a comment explaining that Chain Core 1.2.x generated snapshots without populating versions.